napari/layers/utils/text.py

```diff
@@ -1,242 +1,98 @@
+    size : float
+        Font size of the text, which must be positive. Default value is 12.
+    color : array
+        Font color for the text as an [R, G, B, A] array. Can also be expressed
+        as a string on construction or setting.
+    blending : Blending
+        The blending mode that determines how RGB and alpha values of the layer
+        visual get mixed. Allowed values are 'translucent' and 'additive'.
+        Note that 'opaque` blending is not allowed, as it colors the bounding box
+        surrounding the text, and if given, 'translucent' will be used instead.
+    anchor : Anchor
         The location of the text origin relative to the bounding box.
         Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'.
     translation : np.ndarray
         Offset from the anchor point.
-    color : array
-        Font color for the text
-    size : float
-        Font size of the text. Default value is 12.
-    The blending mode that determines how RGB and
-        alpha values of the layer visual get mixed. Allowed values are
-        {'opaque', 'translucent', and 'additive'}. Note that 'opaque` blending
-        is not recommended, as colors the bounding box surrounding the text.
-    visible : bool
-        Set to true of the text should be displayed.
+    rotation : float
+        Angle of the text elements around the anchor point. Default value is 0.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
-
-    @property
-    def values(self):
-        """np.ndarray: the text values to be displayed"""
-        return self._values
+    values: Array[str, (-1,)] = None
+    visible: bool = True
+    size: PositiveInt = 12
+    color: Array[float, (4,)] = None
+    blending: Blending = Blending.TRANSLUCENT
+    anchor: Anchor = Anchor.CENTER
+    translation: Array[float, (-1,)] = None
+    rotation: float = 0
+    _mode: TextMode = TextMode.NONE
+    _text_format_string: str = None
+
+    def __init__(self, text=None, n_text=None, properties=None, **kwargs):
```

Comment:
I find this `__init__` a little confusing.  I'm wondering if this is here more to preserve the legacy signature?  The docstring kind of makes it sound like I can initialize the values with this field but:

```python
In [1]: from napari.layers.utils.text import TextManager

In [2]: tm = TextManager(text=['hi'])

In [3]: tm
Out[3]: TextManager(values=array([], dtype='<U32'), visible=True, size=12, color=array([0., 1., 1., 1.]), blending='translucent', anchor='center', translation=array([0.]), rotation=0.0)
```

Also, if someone were to use both `values` and `text`, `text` would get silently clobbered, which is perhaps confusing.

Is this how a user would actually initialize `TextManager`?  Or is this how napari "has been" initializing it?

If we have a commonly-used construction signature elsewhere in the codebase, perhaps we should have some `@classmethod` for that specific case?
```python

class TextManager:
    @classmethod
    def from_text(self, text=None, n_text=None, properties=None):
        ...
```

Reply:
Agreed that `__init__` is confusing, and that text/values are confusing. See the comments and discussion above with @kevinyamauchi for a little more context.

Before this PR, the closest legal usage to your example is:

```python
import napari.layers.utils.text
text = napari.layers.utils.text.TextManager(['hi'], n_text=1)
text.values
Out[4]: array([], dtype=float64)
```

so I don't think this PR made the behavior worse, but it also didn't improve things much either (other than the `dtype` of the array).

In general, this PR mostly changes the implementation and tries to keep the behavior as close as possible, with some differences due to exposing `values` as a public, settable attribute.

I like the factory `@classmethod` you proposed and thought about doing something similar. I also thought about factory methods like `from_format` and `from_properties`, which could help better distinguish how an instance of `TextManager` is being used. The implementation of `TextManager` is heavily dependent on what mode it's in, which is something I struggled to follow initially.

I'm happy to try to improve things in this PR, or a follow-up, but I need to think about things a little more.

Reply:
thanks for the details.  I agree with the sentiments that it's not critical to address here, so feel free to merge whenever. (but maybe include a warning in the `__init__` if someone provides both text and values?).  As someone who doesn't use `TextManager` directly a lot, I would definitely be confused by that if I stumbled onto it.